### ISSUE TITLE
Some characters are displaying only after switching windows

### ISSUE THREAD

**Issue by mlopes** [tmux/tmux#928]
I'm using tmux 2.4 on Mac OS X 10.12.4 on iTerm2  3.0.15. I'm using some special characters in my prompt to indicate the state of git repositories. This works well on iTerm2, outside of tmux.
On tmux, when the prompt is presented, the character appears as a blank space:

![screen shot 2017-05-13 at 22 04 21](https://cloud.githubusercontent.com/assets/654/26029226/2cba453a-3828-11e7-9953-2d7cdebf503e.png)
 
When I switch to a different tmux window and then back (example ctrl-b n and then back ctrl-b p), then the character is shown:

![screen shot 2017-05-13 at 22 06 38](https://cloud.githubusercontent.com/assets/654/26029254/763d105c-3828-11e7-911f-e609d1519c6f.png)

From inside tmux TERM is `screen-256color` and outside `xterm-256color`.

```
$ uname -sp && tmux -V && echo $TERM
Darwin i386
tmux 2.4
screen-256color
```
Running 
```
tmux -Ltest kill-server
tmux -vvvv -Ltest -f/dev/null new
```
Seems to show that the problem is intermittent, as shown in the screenshot bellow:

![image](https://cloud.githubusercontent.com/assets/654/26029314/2134bb94-382a-11e7-8fec-fc87970da1e2.png)

Note as the arrow in the prompt only shows up sometimes. This was achieved by simply pressing enter a few times. Outside tmux the arrow will always show (unless the repository state would change). This happens with a few other characters as well.

**Comment by nicm** [comment]
Does this still happen with master?

**Comment by mlopes** [comment]
Just compiled it from master, and was still able to reproduce it

```
$ ./tmux -V
tmux master
```

**Comment by nicm** [comment]
Can you send me logs from reproducing with master please?


On 13 May 2017 10:34 pm, "Marco Lopes" <notifications@github.com> wrote:

Just compiled it from master, and was still able to reproduce it

$ ./tmux -V
tmux master

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/tmux/tmux/issues/928#issuecomment-301276163>, or mute
the thread

**Comment by mlopes** [comment]
Hi, here are the 3 files that are generated when I run tmux with the parameters described above:


[tmux-client-8855.log.zip](https://github.com/tmux/tmux/files/999613/tmux-client-8855.log.zip)
[tmux-out-8857.log.zip](https://github.com/tmux/tmux/files/999614/tmux-out-8857.log.zip)
[tmux-server-8857.log.zip](https://github.com/tmux/tmux/files/999612/tmux-server-8857.log.zip)

**Comment by agsimeonov** [comment]
I am experiencing the exact same issue on tmux 2.4 however everything displays fine and as expected on tmux 2.3

**Comment by nicm** [comment]
This log is from 2.4, not from master:

1494785506.985623 server started (8857): version 2.4, socket /private/tmp/tmux-501/test, protocol 8

**Comment by mlopes** [comment]
ok, so I think I may have figured what was happening. The problem is with version 2.4, but if I use master to connect to a socket that has been created by 2.4 previously, I'll see the problem on master as well.
I removed all the socket files, and ran tmux compiled from master, and the problem seemed to be gone.

**Comment by nicm** [comment]
You need to restart tmux entirely after you upgrade.

**Comment by nicm** [comment]
This should be fixed in 2.5 when it is released probably in about 4 weeks.
